src/Tools/dotnet-monitor/HostBuilder/HostBuilderSettings.cs

```diff
@@ -42,6 +55,22 @@ internal sealed class HostBuilderSettings
                     Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "." + ProductFolderName) :
                     Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), ProductFolderName));
 
+        // Location where extensions are stored by default.
+        // Windows: "%USERPROFILE%\.dotnet\Tools"
+        // Other: "%XDG_CONFIG_HOME%/.dotnet/tools" OR "%HOME%/.dotnet/tools" -> THIS HAS NOT BEEN TESTED YET ON LINUX
+        public static readonly string DotnetToolsExtensionDirectoryPath =
```

Comment:
I don't think this override is necessary; use the test hosting startup to inject an override:
1. Change this to be a service (e.g. IDotnetToolsFileSystem.Path) that the ToolsExtensionRepository will consume.
2. Register this new service with TryAddSingleton (e.g. https://github.com/dotnet/dotnet-monitor/blob/main/src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs#L244)
3. Provide an override implementation in the test hosting startup (e.g. https://github.com/dotnet/dotnet-monitor/blob/main/src/Tests/Microsoft.Diagnostics.Monitoring.Tool.TestHostingStartup/HostingStartup.cs#L19)

Reply:
It may have been a result of my implementation, but this ended up being a bit intrusive. Previously, in `ServiceCollectionExtensions`, we used the `targetExtensionFolder` when we add the extension repository - now, if we're getting that from the `IServiceProvider`, my understanding is that we don't know what that value is when we're configuring our services. The main fallout of this was having to move some checks around for whether the directory exists, as well as adding a null check within `FindExtension` in `ExtensionDiscoverer` in case the `targetExtensionFolder` doesn't exist. I wasn't sure if you were expecting me to need to work around these things as part of the change, or if there's a cleaner way to do it.

Reply:
> The main fallout of this was having to move some checks around for whether the directory exists, as well as adding a null check within `FindExtension` in `ExtensionDiscoverer` in case the `targetExtensionFolder` doesn't exist. I wasn't sure if you were expecting me to need to work around these things as part of the change, or if there's a cleaner way to do it.

I would push that particular directory check into `ToolsExtensionRepository` and have it preemptively fail to return extensions if they don't exist.

Reply:
That was my initial thought too, but when I did that I got an error when the `createDelegate` in `ServiceCollectionExtensions` would attempt to create a `PhysicalFileProvider` using a path that didn't exist. Technically, I guess the responsibility for creating the `IFileProvider` could be pushed down to the `*ExtensionRepository` level as part of `TryFindExtension` if it doesn't already exist, but I'm still concerned about what I referenced in my second comment above.

Reply:
> I'm still concerned about what I referenced in my second comment above.

I can see the individual configuration file notion for the user and shared settings path fairly easily. I'm not sure that would translate to the dotnet tools directory though since it's not a simple "take all json files in this folder" approach. It would have to use the same algorithm that your discovery code uses and likely be implemented as a custom configuration source.

Reply:
> When I did that I got an error when the `createDelegate` in `ServiceCollectionExtensions` would attempt to create a `PhysicalFileProvider` using a path that didn't exist.

You can return a `NullFileProvider` if the directory doesn't exist, so maybe a static method that checks if it exists and then returns wither `PhysicalFileProvider` or `NullFileProvider`.